src/Icicle/Source/PrettyAnnot.hs

```diff
@@ -1,74 +1,133 @@
 {-# LANGUAGE NoImplicitPrelude #-}
 {-# LANGUAGE OverloadedStrings #-}
 {-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE FlexibleContexts  #-}
+{-# LANGUAGE PatternGuards     #-}
 module Icicle.Source.PrettyAnnot (
     PrettyAnnot(..)
   ) where
 
+import                  Icicle.Source.Type
 import                  Icicle.Source.Query
 import                  Icicle.Internal.Pretty
 
 import                  P
 
-import                  Data.List (intersperse)
-
 newtype PrettyAnnot q
  = PrettyAnnot
  { getPrettyAnnot :: q }
 
-instance (Pretty a, Pretty n) => Pretty (PrettyAnnot (Exp a n)) where
+instance (Pretty a, Pretty n) => Pretty (PrettyAnnot (Exp (Annot a n) n)) where
```

Comment:
I was thinking if we just modified the pretty instance for Annot, we wouldn't need overlap?
```
instance {-# don't need a #-} Pretty n => Pretty (Annot a n) where
 pretty a = … pretty (annResult a)

instance Pretty a => Pretty (Exp a n) where
 pretty (Var a n) = annotate a (pretty n)
```